### PR TITLE
Update azure-ai-inference client library to support sending images as part of chat completions

### PR DESCRIPTION
**Pull request by dargilco**
Re-emit from latest TypeSpec to pick up these changes:

- Remove `capacity_type` on result object, as it was removed from the REST API spec 
- Add strongly-typed support for sending images up to the service as part of chat completions request. You will need to use an AI model that supports "vision", like gpt-4-turbo or phi3-v.

Additional changes:

- Add two new chat completions samples, one showing how to send an image using strongly-typed user message, the other showing how to do it by providing a dictionary as input. Both samples use sample1.png that's included in the samples folder.
- Swap sample image names (sample1.png, sample2.png) since I get more interesting results ("describe what's in the image") for the (new) sample1.png.
- Update samples README.md to mention the two new samples
- Update package (root) README.md with link to sample showing how to send an image

Example of sending an image:
```python

    response = client.complete(
        messages=[
            SystemMessage(content="You are an AI assistant that describes images in details."),
            UserMessage(
                content=[
                    TextContentItem(text="What's in this image?"),
                    ImageContentItem(
                        image_url=ImageUrl(
                            url="https://raw.githubusercontent.com/Azure/azure-sdk-for-python/main/sdk/ai/azure-ai-inference/samples/sample1.png",
                            detail=ImageDetailLevel.HIGH,
                        ),
                    ),
                ],
            ),
        ],
    )
```

### CODE REVIEW

File: sdk/ai/azure-ai-inference/azure/ai/inference/models/_models.py [hunk 1]
```diff
@@ -469,6 +469,153 @@ def __init__(self, *args: Any, **kwargs: Any) -> None:  # pylint: disable=useles
...
+        """
+        :param mapping: raw JSON to initialize the model.
+        :type mapping: Mapping[str, Any]
+        """
+
+    def __init__(self, *args: Any, **kwargs: Any) -> None:  # pylint: disable=useless-super-delegation
+        super().__init__(*args, **kwargs)
+
+
+class ImageContentItem(ContentItem, discriminator="image_url"):
```
Comment (johanste) [inline sdk/ai/azure-ai-inference/azure/ai/inference/models/_models.py:506]: Does this allow for inline images/data urls?

File: sdk/ai/azure-ai-inference/azure/ai/inference/models/_models.py [hunk 2]
```diff
@@ -469,6 +469,153 @@ def __init__(self, *args: Any, **kwargs: Any) -> None:  # pylint: disable=useles
...
+        """
+        :param mapping: raw JSON to initialize the model.
+        :type mapping: Mapping[str, Any]
+        """
+
+    def __init__(self, *args: Any, **kwargs: Any) -> None:  # pylint: disable=useless-super-delegation
+        super().__init__(*args, type="image_url", **kwargs)
+
+
+class ImageUrl(_model_base.Model):
```
Comment (johanste) [inline sdk/ai/azure-ai-inference/azure/ai/inference/models/_models.py:544]: This name is misleading - it represents both the url (location) as well as a parameter for how the data is to be handled by the service.

**Review by johanste — APPROVED** [review]

File: sdk/ai/azure-ai-inference/azure/ai/inference/models/_models.py (hunk 1 above)
Comment (dargilco) [reply sdk/ai/azure-ai-inference/azure/ai/inference/models/_models.py:506]: Yes! let me add a sample for sending an image file up.

Comment (dargilco) [reply sdk/ai/azure-ai-inference/azure/ai/inference/models/_models.py:506]: I'll do this in a separate PR.

**Pull request merged**